avaje-nima/src/main/java/io/avaje/nima/AppLifecycle.java

```diff
@@ -0,0 +1,76 @@
+package io.avaje.nima;
+
+/**
+ * Application lifecycle support.
+ */
+public interface AppLifecycle {
+
+  enum Status {
+    STARTING,
+    STARTED,
+    STOPPING,
+    STOPPED
+  }
+
+  interface Event {
+    Status status();
+  }
+
+  interface Callback {
```

Comment:
maybe split these into functional interfaces? I can't say I'm a fan of anonymous classes.

Reply:
yeah, I'll have a look at that (functional interfaces).

This started off with having an `Event` passed to the callback methods - with the thinking that that design would be more robust if something needed to be supplied to the callback (like WebServer). I figured that if we hit that then just change the design and do a major version bump.

but yeah, might try the using functional interfaces and see what it looks like.

Reply:
Got both now